### PR TITLE
docs: Cover customizing settings from a custom Docker image

### LINKED ISSUE #8358: Customize settings in Docker image without touching the data volume
### Describe the issue

How can I create a Dockerfile that inherits from the official Weblate Docker image, and defines additional settings that cannot be defined through environment variables, like `INSTALLED_APPS`, *without* relaying on the data volume?

[The documentation covers the data volume approach](https://docs.weblate.org/en/latest/admin/install/docker.html#custom-configuration-files), which was added to resolve https://github.com/WeblateOrg/weblate/issues/3000, but I believe this is not an ideal approach for changes that are meant to be independent of Docker volumes, i.e. changes that are not only intended for a single instance of Weblate, and that we want to track through a version control system like Git.

That is, I want to be able to have 2 environments (e.g. production and staging), modify my custom Dockerfile that inherits from the official Weblate Docker image to add some static files, and be able to get the changes into both environments (that have independent data volumes) by rebuilding my Docker images (i.e. running `docker-compose up -d` to restart the app on each environment with the latest Dockerfile changes), instead of having to edit the contents of the data volume which is not under version control.

I imagine I could edit the Dockerfile entry point to force the execution of some script that would copy the files previously copied into somewhere in the image into the corresponding mounted volume folder on start (maybe with rsync -delete), but it feels hacky and risky to rewrite volume data on every container restart, instead of copying the files only once during the Docker image building process.

I have also pondered overwriting files under `/usr/local/lib/python3.10/site-packages/`, but that feels even more hacky and risky.

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

_No response_

### Expected behavior

_No response_

### Screenshots

_No response_

### Exception traceback

_No response_

### How do you run Weblate?

Docker container

### Weblate versions

_No response_

### Weblate deploy checks

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Gallaecio**
## Proposed changes

Cover in the documentation how to customize settings when using Docker, *without* using the data volume.

Resolves #8358.

While I think there may be valid use cases for the data volume approach, and it is in many ways simpler, I think the alternative approach proposed here may be preferable in some scenarios.

However, no hard feelings if this change gets rejected. I can think of a few valid reasons for rejection, such as this not being a common scenario (I thought it would be, but it does not look like it is), or how bad it plays with https://github.com/WeblateOrg/docker-compose.

Regardless of whether these changes are merged or rejected, I think we can close #8358 along with this merge request.

## Checklist

- [x] Lint and unit tests pass locally with my changes. (only changed docs, `make html` works without warnings)
- [ ] ~I have added tests that prove my fix is effective or that my feature works~. (does not apply)
- [x] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

### CODE REVIEW

**Comment by nijel** [comment]
I think this makes sense to include, though it is definitely not the most common use case.

**Comment by nijel** [comment]
Looks good, but I'd like somebody else to read that as well.

Rendered docs: https://weblate--8365.org.readthedocs.build/en/8365/admin/install/docker.html#configuration-beyond-environment-variables

File: docs/admin/customize.rst [hunk 1]
```diff
@@ -25,11 +25,17 @@ If you are not familiar with Python, you might want to look into `Python For
...
-To write some custom Python code (called a module), a
-place to store it is needed, either in the system path (usually something like
+To write a file with custom Python code (called a module), a place to store it
+is needed, either in the system path (usually something like
 :file:`/usr/lib/python3.9/site-packages/`) or in the Weblate directory, which
 is also added to the interpreter search path.
 
+.. versionadded:: 3.8-5
+
+   You can place Python modules in :file:`/app/data/python/` (see
```
[the remark this answers is not in the document]
Comment (Gallaecio) [reply docs/admin/customize.rst]: Does https://github.com/WeblateOrg/weblate/pull/8365/commits/5f59919a5f827144efac37ff7e752a96592f16a3 work? :crossed_fingers:

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**